src/base/bittorrent/session.cpp

```diff
@@ -5158,7 +5166,8 @@ void Session::handleSessionStatsAlert(const lt::session_stats_alert *p)
     const auto calcRate = [interval](const qint64 previous, const qint64 current) -> qint64
     {
         Q_ASSERT(current >= previous);
-        return ((current - previous) / interval);
+        Q_ASSERT(interval >= 0);
```

Comment:
It seems this check is redundant. This condition is already guarded by the zero interval check.
Or you can add an comment about "denominator should not be 0".

Reply:
> Or you can add an comment about "denominator should not be 0".

This ASSERT says the same thing.
In addition, it will warn about incorrect use of this function if the external code is reformatted and there will be no such check.